forui/lib/src/widgets/header/header.dart

```diff
@@ -8,158 +8,105 @@ import 'package:forui/src/foundation/tappable.dart';
-/// navigation stack.
+/// A header contains the page's title and actions.
+
 ///
 /// See:
 /// * https://forui.dev/docs/header for working examples.
 /// * [FHeaderStyle] for customizing a header's appearance.
-class FHeader extends StatelessWidget {
-  /// The header's style. Defaults to [FThemeData.headerStyle].
-  final FHeaderStyle? style;
+sealed class FHeader extends StatelessWidget {
+  const FHeader._({super.key});
 
-  /// The title, aligned to the left.
+  /// Creates a header.
   ///
-  /// ## Contract:
-  /// Throws [AssertionError]:
-  /// * if [title] and [rawTitle] are both not null
-  /// * if [title] and [rawTitle] are both null
-  final String? title;
-
-  /// The title, aligned to the left.
+  /// It is typically used on pages at the root of the navigation stack.
+  const factory FHeader({
+    required Widget title,
+    FRootHeaderStyle? style,
+    List<Widget> actions,
+    Key? key,
+  }) = _FRootHeader;
+
+  /// Creates a nested header.
   ///
-  /// ## Contract:
-  /// Throws [AssertionError]:
-  /// * if [title] and [rawTitle] are both not null
-  /// * if [title] and [rawTitle] are both null
-  final Widget? rawTitle;
-
-  /// The actions, aligned to the right. Defaults to an empty list.
-  final List<Widget> actions;
-
-  /// Creates a [FHeader].
-  ///
-  /// ## Contract:
-  /// Throws [AssertionError]:
-  /// * if [title] and [rawTitle] are both not null
-  /// * if [title] and [rawTitle] are both null
-  const FHeader({
-    this.style,
-    this.title,
-    this.rawTitle,
-    this.actions = const [],
-    super.key,
-  }) : assert((title != null) ^ (rawTitle != null), 'title or rawTitle must be provided, but not both.');
-
-  @override
-  Widget build(BuildContext context) {
-    final style = this.style ?? context.theme.headerStyle;
-
-    final title = switch ((this.title, rawTitle)) {
-      (final String title, _) => Text(title),
-      (_, final Widget title) => title,
-      _ => const Placeholder(),
-    };
-
-    return SafeArea(
-      bottom: false,
-      child: Padding(
-        padding: style.padding,
-        child: Row(
-          mainAxisAlignment: MainAxisAlignment.spaceBetween,
-          children: [
-            Expanded(
-              child: DefaultTextStyle.merge(
-                overflow: TextOverflow.fade,
-                maxLines: 1,
-                softWrap: false,
-                style: style.titleTextStyle,
-                child: title,
-              ),
-            ),
-            Row(children: actions),
-          ],
-        ),
-      ),
-    );
-  }
-
-  @override
-  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
-    super.debugFillProperties(properties);
-    properties
-      ..add(DiagnosticsProperty('style', style))
-      ..add(StringProperty('title', title))
-      ..add(IterableProperty('actions', actions));
-  }
+  /// It is typically used on pages NOT at the root of the navigation stack.
+  const factory FHeader.nested({
```

Comment:
Wait, you can actually make factory constructors const? TIL

Reply:
Yep